File: src/main/java/com/github/vootelerotov/intellij/chill/pill/transformer/MethodVisitorAdapter.java
package com.github.vootelerotov.intellij.chill.pill.transformer;

import org.objectweb.asm.*;

class MethodVisitorAdapter extends MethodVisitor  {

  private final MethodVisitor delegate;

  public MethodVisitorAdapter(MethodVisitor delegate) {
    super(Opcodes.ASM9);

    this.delegate = delegate;
  }

  @Override
  public void visitParameter(String name, int access) {
    delegate.visitParameter(name, access);
  }

  @Override
  public AnnotationVisitor visitAnnotationDefault() {
    return delegate.visitAnnotationDefault();
  }

  @Override
  public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
    return delegate.visitAnnotation(descriptor, visible);
  }

  @Override
  public AnnotationVisitor visitTypeAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
    return delegate.visitTypeAnnotation(typeRef, typePath, descriptor, visible);
  }

  @Override
  public void visitAnnotableParameterCount(int parameterCount, boolean visible) {
    delegate.visitAnnotableParameterCount(parameterCount, visible);
  }

  @Override
  public AnnotationVisitor visitParameterAnnotation(int parameter, String descriptor, boolean visible) {
    return delegate.visitParameterAnnotation(parameter, descriptor, visible);
  }

  @Override
  public void visitAttribute(Attribute attribute) {
    delegate.visitAttribute(attribute);
  }

  @Override
  public void visitCode() {
    delegate.visitCode();
  }

  @Override
  public void visitFrame(int type, int numLocal, Object[] local, int numStack, Object[] stack) {
    delegate.visitFrame(type, numLocal, local, numStack, stack);
  }

  @Override
  public void visitInsn(int opcode) {
    delegate.visitInsn(opcode);
  }

  @Override
  public void visitIntInsn(int opcode, int operand) {
    delegate.visitIntInsn(opcode, operand);
  }

  @Override
  public void visitVarInsn(int opcode, int var) {
    delegate.visitVarInsn(opcode, var);
  }

  @Override
  public void visitTypeInsn(int opcode, String type) {
    delegate.visitTypeInsn(opcode, type);
  }

  @Override
  public void visitFieldInsn(int opcode, String owner, String name, String descriptor) {
    delegate.visitFieldInsn(opcode, owner, name, descriptor);
  }

  @Override
  public void visitMethodInsn(int opcode, String owner, String name, String descriptor) {
    delegate.visitMethodInsn(opcode, owner, name, descriptor);
  }

  @Override
  public void visitMethodInsn(int opcode, String owner, String name, String descriptor, boolean isInterface) {
    delegate.visitMethodInsn(opcode, owner, name, descriptor, isInterface);
  }

  @Override
  public void visitInvokeDynamicInsn(String name, String descriptor, Handle bootstrapMethodHandle, Object... bootstrapMethodArguments) {
    delegate.visitInvokeDynamicInsn(name, descriptor, bootstrapMethodHandle, bootstrapMethodArguments);
  }

  @Override
  public void visitJumpInsn(int opcode, Label label) {
    delegate.visitJumpInsn(opcode, label);
  }

  @Override
  public void visitLabel(Label label) {
    delegate.visitLabel(label);
  }

  @Override
  public void visitLdcInsn(Object value) {
    delegate.visitLdcInsn(value);
  }

  @Override
  public void visitIincInsn(int var, int increment) {
    delegate.visitIincInsn(var, increment);
  }

  @Override
  public void visitTableSwitchInsn(int min, int max, Label dflt, Label... labels) {
    delegate.visitTableSwitchInsn(min, max, dflt, labels);
  }

  @Override
  public void visitLookupSwitchInsn(Label dflt, int[] keys, Label[] labels) {
    delegate.visitLookupSwitchInsn(dflt, keys, labels);
  }

  @Override
  public void visitMultiANewArrayInsn(String descriptor, int numDimensions) {
    delegate.visitMultiANewArrayInsn(descriptor, numDimensions);
  }

  @Override
  public AnnotationVisitor visitInsnAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
    return delegate.visitInsnAnnotation(typeRef, typePath, descriptor, visible);
  }

  @Override
  public void visitTryCatchBlock(Label start, Label end, Label handler, String type) {
    delegate.visitTryCatchBlock(start, end, handler, type);
  }

  @Override
  public AnnotationVisitor visitTryCatchAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
    return delegate.visitTryCatchAnnotation(typeRef, typePath, descriptor, visible);
  }

  @Override
  public void visitLocalVariable(String name, String descriptor, String signature, Label start, Label end, int index) {
    delegate.visitLocalVariable(name, descriptor, signature, start, end, index);
  }

  @Override
  public AnnotationVisitor visitLocalVariableAnnotation(int typeRef, TypePath typePath, Label[] start, Label[] end, int[] index, String descriptor, boolean visible) {
    return delegate.visitLocalVariableAnnotation(typeRef, typePath, start, end, index, descriptor, visible);
  }

  @Override
  public void visitLineNumber(int line, Label start) {
    delegate.visitLineNumber(line, start);
  }

  @Override
  public void visitMaxs(int maxStack, int maxLocals) {
    delegate.visitMaxs(maxStack, maxLocals);
  }

  @Override
  public void visitEnd() {
    delegate.visitEnd();
  }
}

File: src/main/kotlin/com/github/vootelerotov/intellij/chill/pill/transformer/MethodVisitorAdapter.kt
package com.github.vootelerotov.intellij.chill.pill.transformer

import org.objectweb.asm.*

open class MethodVisitorAdapter(val delegate: MethodVisitor) : MethodVisitor(Opcodes.ASM9) {

  override fun visitParameter(name: String?, access: Int) {
    delegate.visitParameter(name, access)
  }

  override fun visitAnnotationDefault(): AnnotationVisitor {
    return delegate.visitAnnotationDefault()
  }

  override fun visitAnnotation(descriptor: String?, visible: Boolean): AnnotationVisitor {
    return delegate.visitAnnotation(descriptor, visible)
  }

  override fun visitTypeAnnotation(
    typeRef: Int,
    typePath: TypePath?,
    descriptor: String?,
    visible: Boolean
  ): AnnotationVisitor {
    return delegate.visitTypeAnnotation(typeRef, typePath, descriptor, visible)
  }

  override fun visitAnnotableParameterCount(parameterCount: Int, visible: Boolean) {
    delegate.visitAnnotableParameterCount(parameterCount, visible)
  }

  override fun visitParameterAnnotation(parameter: Int, descriptor: String?, visible: Boolean): AnnotationVisitor {
    return delegate.visitParameterAnnotation(parameter, descriptor, visible)
  }

  override fun visitAttribute(attribute: Attribute?) {
    delegate.visitAttribute(attribute)
  }

  override fun visitCode() {
    delegate.visitCode()
  }

  override fun visitFrame(type: Int, numLocal: Int, local: Array<out Any>?, numStack: Int, stack: Array<out Any>?) {
    delegate.visitFrame(type, numLocal, local, numStack, stack)
  }

  override fun visitInsn(opcode: Int) {
    delegate.visitInsn(opcode)
  }

  override fun visitIntInsn(opcode: Int, operand: Int) {
    delegate.visitIntInsn(opcode, operand)
  }

  override fun visitVarInsn(opcode: Int, `var`: Int) {
    delegate.visitVarInsn(opcode, `var`)
  }

  override fun visitTypeInsn(opcode: Int, type: String?) {
    delegate.visitTypeInsn(opcode, type)
  }

  override fun visitFieldInsn(opcode: Int, owner: String?, name: String?, descriptor: String?) {
    delegate.visitFieldInsn(opcode, owner, name, descriptor)
  }

  override fun visitMethodInsn(opcode: Int, owner: String?, name: String?, descriptor: String?) {
    delegate.visitMethodInsn(opcode, owner, name, descriptor)
  }

  override fun visitMethodInsn(opcode: Int, owner: String?, name: String?, descriptor: String?, isInterface: Boolean) {
    delegate.visitMethodInsn(opcode, owner, name, descriptor, isInterface)
  }

  override fun visitInvokeDynamicInsn(
    name: String?,
    descriptor: String?,
    bootstrapMethodHandle: Handle?,
    vararg bootstrapMethodArguments: Any?
  ) {
    delegate.visitInvokeDynamicInsn(name, descriptor, bootstrapMethodHandle, *bootstrapMethodArguments)
  }

  override fun visitJumpInsn(opcode: Int, label: Label?) {
    delegate.visitJumpInsn(opcode, label)
  }

  override fun visitLabel(label: Label?) {
    delegate.visitLabel(label)
  }

  override fun visitLdcInsn(value: Any?) {
    delegate.visitLdcInsn(value)
  }

  override fun visitIincInsn(`var`: Int, increment: Int) {
    delegate.visitIincInsn(`var`, increment)
  }

  override fun visitTableSwitchInsn(min: Int, max: Int, dflt: Label?, vararg labels: Label?) {
    delegate.visitTableSwitchInsn(min, max, dflt, *labels)
  }

  override fun visitLookupSwitchInsn(dflt: Label?, keys: IntArray?, labels: Array<out Label>?) {
    delegate.visitLookupSwitchInsn(dflt, keys, labels)
  }

  override fun visitMultiANewArrayInsn(descriptor: String?, numDimensions: Int) {
    delegate.visitMultiANewArrayInsn(descriptor, numDimensions)
  }

  override fun visitInsnAnnotation(
    typeRef: Int,
    typePath: TypePath?,
    descriptor: String?,
    visible: Boolean
  ): AnnotationVisitor {
    return delegate.visitInsnAnnotation(typeRef, typePath, descriptor, visible)
  }

  override fun visitTryCatchBlock(start: Label?, end: Label?, handler: Label?, type: String?) {
    delegate.visitTryCatchBlock(start, end, handler, type)
  }

  override fun visitTryCatchAnnotation(
    typeRef: Int,
    typePath: TypePath?,
    descriptor: String?,
    visible: Boolean
  ): AnnotationVisitor {
    return delegate.visitTryCatchAnnotation(typeRef, typePath, descriptor, visible)
  }

  override fun visitLocalVariable(
    name: String?,
    descriptor: String?,
    signature: String?,
    start: Label?,
    end: Label?,
    index: Int
  ) {
    delegate.visitLocalVariable(name, descriptor, signature, start, end, index)
  }

  override fun visitLocalVariableAnnotation(
    typeRef: Int,
    typePath: TypePath?,
    start: Array<out Label>?,
    end: Array<out Label>?,
    index: IntArray?,
    descriptor: String?,
    visible: Boolean
  ): AnnotationVisitor {
    return delegate.visitLocalVariableAnnotation(typeRef, typePath, start, end, index, descriptor, visible)
  }

  override fun visitLineNumber(line: Int, start: Label?) {
    delegate.visitLineNumber(line, start)
  }

  override fun visitMaxs(maxStack: Int, maxLocals: Int) {
    delegate.visitMaxs(maxStack, maxLocals)
  }

  override fun visitEnd() {
    delegate.visitEnd()
  }
}